targets/TARGET_Silicon_Labs/TARGET_EFM32/pwmout_api.c

```diff
@@ -334,24 +341,34 @@ void pwmout_period_us(pwmout_t *obj, int us)
     pwmout_period(obj, us / 1000000.0f);
 }
 
+int pwmout_read_period_us(pwmout_t *obj)
+{
+    return TIMER_TopGet(PWM_TIMER) / REFERENCE_FREQUENCY * 1000000;
+}
+
 void pwmout_pulsewidth(pwmout_t *obj, float seconds)
 {
-    uint16_t width_cycles = (uint16_t) (((float) (REFERENCE_FREQUENCY >> pwm_prescaler_div)) * seconds);
+    uint16_t width_cycles = (uint16_t)(((float)(REFERENCE_FREQUENCY >> pwm_prescaler_div)) * seconds);
     TIMER_CompareBufSet(PWM_TIMER, obj->channel, width_cycles);
 }
 
 void pwmout_pulsewidth_ms(pwmout_t *obj, int ms)
 {
-    uint16_t width_cycles = (uint16_t) (((REFERENCE_FREQUENCY >> pwm_prescaler_div) * ms) / 1000);
+    uint16_t width_cycles = (uint16_t)(((REFERENCE_FREQUENCY >> pwm_prescaler_div) * ms) / 1000);
     TIMER_CompareBufSet(PWM_TIMER, obj->channel, width_cycles);
 }
 
 void pwmout_pulsewidth_us(pwmout_t *obj, int us)
 {
-    uint16_t width_cycles = (uint16_t) (((uint64_t)(REFERENCE_FREQUENCY >> pwm_prescaler_div) * (uint64_t)us) / 1000000UL);
+    uint16_t width_cycles = (uint16_t)(((uint64_t)(REFERENCE_FREQUENCY >> pwm_prescaler_div) * (uint64_t)us) / 1000000UL);
     TIMER_CompareBufSet(PWM_TIMER, obj->channel, width_cycles);
 }
 
+int pwmout_read_pulsewidth_us(pwmout_t *obj)
+{
+    return TIMER_CaptureGet(PWM_TIMER, obj->channel) / REFERENCE_FREQUENCY * 1000000;
```

Comment:
Same here. Order of operations needs to be specific in order to account for integer math.